tests/ConfigurationSchemaGenerator.Tests/GeneratorTests.cs

```diff
@@ -51,56 +118,1110 @@ public void ShouldRemoveIndentation(string value, string expected)
+          "properties": {
+            "ChildValue": {
+              "type": "string"
+            },
+            "Parent": {}
+          }
+        }
+        """)]
+#if NETCOREAPP
+    [InlineData("Int128",
+        """
+        "type": "integer"
+        """)]
+    [InlineData("UInt128",
+        """
+        "type": "integer"
+        """)]
+    [InlineData("Half",
+        """
+        "type": [
+          "number",
+          "string"
+        ]
+        """)]
+    [InlineData("DateOnly",
+        """
+        "type": "string"
+        """)]
+    [InlineData("TimeOnly",
+        """
+        "type": "string"
+        """)]
+#endif
+    public void ShouldConvertTypes(string propertyType, string expected)
+    {
+        var source =
+            $$"""
+            using System.Globalization;
+            using System.Security.Cryptography.X509Certificates;
+
+            [assembly: Aspire.ConfigurationSchema("TestComponent", typeof(TestSettings), ["TestProperty:ClientCertificate"])]
+
+            public class TestSettings
+            {
+                public {{propertyType}} TestProperty { get; set; }
+            }
+
+            public record ChildType
+            {
+                public string? ChildValue { get; set; }
+
+                // Simple read-only property, ignored.
+                public bool HasParent => Parent != null;
+
+                // Circular reference, included but without type.
+                public ChildType? Parent { get; set; }
+
+                // Ignored via exclusionPaths.
+                public X509Certificate2? ClientCertificate { get; set; }
+            }
+            """;
+
+        ImmutableArray<MetadataReference> references = [
+            MetadataReference.CreateFromFile(typeof(Uri).Assembly.Location),
+            MetadataReference.CreateFromFile(typeof(X509Certificate2).Assembly.Location)
+        ];
+        var schema = GenerateSchemaFromCode(source, references);
+
+        AssertIsJson(schema,
+          $$"""
+          {
+            "type": "object",
+            "properties": {
+              "TestComponent": {
+                "type": "object",
+                "properties": {
+                  "TestProperty": {
+                    {{expected}}
+                  }
+                }
+              }
+            }
+          }
+          """);
+    }
+
+    [Fact]
+    public void ShouldAllowGetOnlyObject()
+    {
+        var source =
+            """
+            [assembly: Aspire.ConfigurationSchema("TestComponent", typeof(TestSettings))]
+
+            public class TestSettings
+            {
+                public JsonFormattingOptions FormatSettings { get; } = new();
+            }
+
+            public class JsonFormattingOptions
+            {
+                public bool WriteIndented { get; set; }
+            }
+            """;
+
+        var schema = GenerateSchemaFromCode(source, []);
+
+        Assert.Contains("FormatSettings", schema);
```

Comment:
In general, get-only properties are excluded in the schema. But collections and non-primitive types are exceptions to that. This test verifies the property gets emitted, even if the declared type is get-only.

I didn't assert on the full tree, because most of its content isn't relevant here. This way, future refactorings of other tree parts won't require "fixing" this test.

Reply:
But it doesn't catch other bugs. It also doesn't validate that "FormatSettings" gets put in the right location. Since we are validating the full tree everywhere else, it feels like we should just do the same here. It gets us better test coverage.

Reply:
Existing tests already cover all other concerns. It's _duplicating_ test coverage, not making it better. But I've added it anyway.